server/src/main/scala/com/cloudera/livy/utils/SparkYarnApp.scala

```diff
@@ -50,6 +58,47 @@ object SparkYarnApp extends Logging {
 
   private def getYarnPollInterval(livyConf: LivyConf): FiniteDuration =
     livyConf.getTimeAsMs(LivyConf.YARN_POLL_INTERVAL) milliseconds
+
+  private val appType = Set("SPARK").asJava
+
+  private val leakagedAppTags = new java.util.concurrent.ConcurrentHashMap[String, Long]()
```

Comment:
I tried it, but it looks like it is a little complicated than I expect. Because it may affect the code of batch session recovering. The remaining issue is that the leaked session can not be cleanup after recovery, I think this is not critical, so we can do it in a follow up ticket, what do you think ?